File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        int n = s.size();
        stack<char> st;

        for(int i=0; i<n; ++i){
            char c = s[i];
            if(c=='(' or c=='{' or c=='['){
                st.push(c);
            }
            else{
                if(st.empty())
                    return false;
                char t = st.top();
                if((c==')' and t!='(') or (c=='}' and t!='{') or (c==']' and t!='['))
                    return false;
                 st.pop();
            }
        }
        if(st.empty())
            return true;
        return false;
    }
};

File: 20-valid-parentheses/valid-parentheses.java
import java.util.*;

class Solution {
    public boolean isValid(String s) {
        int n = s.length();

        Stack<Character> st = new Stack<>();

        for (int i = 0; i < n; ++i) {
            char c = s.charAt(i);
            if (c == '(' || c == '{' || c == '[') {
                st.push(c);
            } else {
                if (st.empty())
                    return false;
                char t = st.peek();

                if ((c == ')' && t != '(') || (c == '}' && t != '{') || (c == ']' && t != '['))
                    return false;
                st.pop();
            }
        }
        return st.empty();
    }
}